runtime/lua/man.lua

```diff
@@ -201,6 +201,7 @@ local function highlight_man_page()
   api.nvim_buf_set_lines(0, 0, -1, false, lines)
 
   for _, hl in ipairs(hls) do
+    --- @diagnostic disable-next-line: deprecated
```

Comment:
For my own learning: Why do we need to use the deprecated function here?

Reply:
Same comment for all places where we use `nvim_buf_add_highlight`

Reply:
We don't have to, but whoever deprecated it didn't update the internal uses. It's not going to be removed until 1.0 so it's low risk anyway.